lib/squadster/schedules/services/create_lesson.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Squadster.Schedules.Services.CreateLesson do
+  alias Squadster.Repo
+  alias Squadster.Schedules.Lesson
+
+  def call(timetable, %{index: index} = args) do
+    timetable.lessons
+    |> Enum.find(fn lesson -> lesson.index == index end)
+    |> case do
+      nil -> create(args)
+      _lesson -> {:error, "Lesson with index #{index} already exists"}
+    end
+  end
+
+  def call(timetable, args), do: create(Map.merge(args, %{index: get_last_index(timetable.lessons)}))
```

Comment:
It seems messy, maybe one-line function is not the best idea here?

Reply:
I will replace it with pipe and make it multiline